src/raftstore/store/fsm/peer.rs

```diff
@@ -1035,7 +1029,21 @@ impl<T: Transport, C: PdClient> Store<T, C> {
             }
 
             new_peer.register_delegates();
-            self.region_peers.insert(new_region_id, new_peer);
+
+            if let Some(exist_peer) = self.region_peers.insert(new_region_id, new_peer) {
+                // Suppose a new node is added by conf change and the snapshot comes slowly.
+                // Then, the region splits and the first vote message comes to the new node
+                // before the old snapshot, which will create an uninitialized peer on the
+                // store. After that, the old snapshot comes, followed with a split proposal.
```

Comment:
> followed with a split proposal

why another split proposal?

Reply:
It's just the last split proposal, not a new split. Seems `followed with the last split proposal` is better.